gnucash/gnome-utils/gnc-window.c

```diff
@@ -139,6 +140,33 @@ gnc_window_set_status (GncWindow *window, GncPluginPage *page,
  * bad from C, but also has to be done in Scheme.
  */
 static GncWindow *progress_bar_hack_window = NULL;
+static gulong progressbar_handler;
+static ProgressbarHandler *escape_handler_cb;
+
+static gboolean
+progressbar_kp (GtkWidget *widget, GdkEventKey *event, gpointer data)
+{
+    if (event->length == 0) return FALSE;
+
+    switch (event->keyval)
+    {
+    case GDK_KEY_Escape:
+        if (escape_handler_cb)
+        {
+            (*escape_handler_cb)(widget);
```

Comment:
instead of the `escape_handler_cb` being customised by the caller, we can also customise the whole g_signal handler. Would that be better?

Reply:
> instead of the escape_handler_cb being customised by the caller, we can also customise the whole g_signal handler. Would that be better?

Which signal and customize how? Would the caller pass in a `GCallback` that some gnc-window function would connect to the signal or would the caller connect directly?

Reply:
> Would the caller pass in a GCallback that some gnc-window function would connect to the signal

yes that'd be the idea. I have since then added 2 more commits which avoids augmenting progressbar and the scrubber itself creates the signal. This sounds cleaner...